pkg/agent/agent_test.go

```diff
@@ -708,3 +712,261 @@ func TestRestorePortConfigs(t *testing.T) {
 		})
 	}
 }
+
+func TestSetOVSDatapath(t *testing.T) {
+	tests := []struct {
+		name          string
+		expectedCalls func(m *ovsconfigtest.MockOVSBridgeClient)
+		expectedErr   string
+	}{
+		{
+			name: "fail to read OVS bridge other_config",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(nil, ovsconfig.InvalidArgumentsError("failed to read OVS bridge other_config"))
```

Comment:
`NewTransactionError` should make more sense here, given there is even no argument needed for this function.